contracts/WooracleV2.sol

```diff
@@ -202,57 +217,101 @@ contract WooracleV2 is InitializableOwnable, IWooracleV2 {
         clFeasible[base] = feasible;
     }
 
-    /* ----- Private Functions ----- */
-
-    function _setState(
-        address base,
-        uint128 newPrice,
-        uint64 newSpread,
-        uint64 newCoeff
-    ) private {
-        TokenInfo storage info = infos[base];
-        info.price = uint64(newPrice);
-        info.spread = newSpread;
-        info.coeff = newCoeff;
-    }
-
-    function cloPrice(address base) external view override returns (uint256 priceNow, uint256 timestampNow) {
-        return (0, 0);
+    function cloPrice(address base) external view override returns (uint256, uint256) {
+        return _refPrice(base, quoteToken);
     }
 
     function isWoFeasible(address base) external view override returns (bool) {
-        return true;
+        uint256 price = uint256(infos[base].price);
+        return price != 0 && block.timestamp <= (timestamp + staleDuration);
     }
 
-    function woSpread(address base) external view override returns (uint256) {
-        return 0;
+    function woSpread(address base) external view override returns (uint64) {
+        return infos[base].spread;
     }
 
-    function woCoeff(address base) external view override returns (uint256) {
-        return 0;
+    function woCoeff(address base) external view override returns (uint64) {
+        return infos[base].coeff;
     }
 
     // Wooracle price of the base token
-    function woPrice(address base) external view override returns (uint256 priceNow, uint256 timestampNow) {
-        return (0, 0);
+    function woPrice(address base) external view override returns (uint128 price, uint256 priceTimestamp) {
+        price = infos[base].price;
+        priceTimestamp = timestamp;
     }
 
     function woState(address base)
         external
         view
         override
         returns (
-            uint256 priceNow,
-            uint256 spreadNow,
-            uint256 coeffNow,
-            uint256 timestampNow
+            uint128 priceNow,
+            uint64 spreadNow,
+            uint64 coeffNow,
+            uint256 priceTimestamp
         )
     {
-        return (0, 0, 0, 0);
+        TokenInfo storage info = infos[base];
+        priceNow = info.price;
+        spreadNow = info.spread;
+        coeffNow = info.coeff;
+        priceTimestamp = timestamp;
     }
 
     function cloAddress(address base) external view override returns (address clo) {
-        clo = 0x02Bfe714e78E2Ad1bb1C2beE93eC8dc5423B66d4;
-        return clo;
+        clo = clOracles[base].oracle;
+    }
+
+    /* ----- Private Functions ----- */
+
+    function _initclOracles() private {
```

Comment:
Seems like not being used any where in this file?

Reply:
Removed